src/main/scala/org/broadinstitute/dsde/workbench/avram/db/EntityComponent.scala

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.dsde.workbench.avram.db
+
+
+import AvramPostgresProfile.api._
+//import org.broadinstitute.dsde.workbench.avram.api.Entity
+import spray.json._
+
+import scala.concurrent.ExecutionContext
+
+case class EntityRecord(id: Long, name: String, collection: Long, entityBody: JsValue)
+
+trait EntityComponent extends AvramComponent  {
+  this: CollectionComponent =>
+
+  class EntityTable(tag: Tag) extends Table[EntityRecord](tag, "entity") {
+    def id =             column[Long]      ("id", O.PrimaryKey, O.AutoInc)
+    def name =           column[String]    ("name", O.Length(254))
+    def collection =     column[Long]      ("collection")
+    def entityBody =     column[JsValue]   ("entityBody", O.Length(254), O.SqlType("JSONB"))
+
+    def collectionForeignKey = foreignKey("FK_COLLECTION", collection, collectionQuery)(_.id)
+    def uniqueKey = index("IDX_NAME_COLLECTION_UNIQUE", (name, collection), unique = true)
+
+    def * = (id, name, collection, entityBody) <> (EntityRecord.tupled, EntityRecord.unapply)
+  }
+
+  object entityQuery extends TableQuery(new EntityTable(_)) {
+
+    def save(name: String, collection: Long, entityBody: JsValue): DBIO[Int] = {
+      entityQuery += EntityRecord(0, name, collection, entityBody)
+    }
```

Comment:
we can use JsValue for the JSONB. The funny thing is that when I pass it up to Cloud Endpoints in a case class
like so: `Entity(....@BeanProperty entityBody: JsValue), it does know what to do with a JsValue so a get returns:

{
 "name": "entityName1",
 "collection": "collection1",
  "entityBody": { }
}

where the entityBody is empty. This doesn't happen if the JsValue is toStringed.

Reply:
Yeah that's not ideal, not sure of the solution though..

Reply:
I'm thinking we might want to serialize our responses into JSON strings ourselves (using circe) instead of trying to get Endpoints Framework to do it for us. That might help with this... especially if we switch from spray-json's `JsValue` to circe's `Json`. I just haven't had a chance to experiment with this yet.